### PR TITLE
Embedding JS dependencies as Go code

### LINKED ISSUE #545: Make k6 builds portable by default
Builds of `k6` created manually via `go get -u github.com/loadimpact/k6` or `go build -a github.com/loadimpact/k6` refuse to run on other machines that don't have the k6 code checked out exactly where it was on the build machine. That's the error:
```
panic: could not locate box "lib"

goroutine 1 [running]:
github.com/loadimpact/k6/vendor/github.com/GeertJohan/go%2erice.MustFindBox(0xc331c9, 0x3, 0x8deabf)
	/go/path/on/system/src/github.com/loadimpact/k6/vendor/github.com/GeertJohan/go.rice/box.go:110 +0x97
```
Even the following is enough to reproduce the issue:
```sh
go get -u github.com/loadimpact/k6
"$GOPATH/bin/k6" # this runs without issues
mv "$GOPATH/src/github.com/loadimpact/k6" "$GOPATH/src/github.com/loadimpact/k6-moved"
"$GOPATH/bin/k6" # this fails
```
Looks like some of the static resources are not properly embedded...
Reported by [vtanakas on slack](https://k6io.slack.com/archives/C3E688J8Z/p1521315432000121)

### PR DESCRIPTION
**Pull request by luizbafilho**
closes #545

### CODE REVIEW

File: js/bundle_test.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func TestNewBundle(t *testing.T) {
 	})
 	t.Run("Invalid", func(t *testing.T) {
 		_, err := getSimpleBundle("/script.js", "\x00")
-		assert.EqualError(t, err, "SyntaxError: /script.js: Unexpected character '\x00' (1:0)\n> 1 | \x00\n    | ^ at <eval>:2:26853(114)")
+		assert.EqualError(t, err, "SyntaxError: /script.js: Unexpected character '\x00' (1:0)\n> 1 | \x00\n    | ^ at <eval>:2:28542(114)")
```
Comment (na--) [inline js/bundle_test.go]: Depending on the JS line number would force us to fix this test every time we update the embedded libs...
I think that using something like `assert.Contains(t, err.Error(), "SyntaxError: /script.js: Unexpected character '\x00' (1:0)\n")` would be good enough and save us some work in the future.

File: js/initcontext_test.go [hunk 2]
```diff
@@ -118,7 +118,7 @@ func TestInitContextRequire(t *testing.T) {
 				Filename: "/script.js",
 				Data:     []byte(`import "/file.js"; export default function() {}`),
 			}, fs, lib.RuntimeOptions{})
-			assert.EqualError(t, err, "SyntaxError: /file.js: Unexpected character '\x00' (1:0)\n> 1 | \x00\n    | ^ at <eval>:2:26853(114)")
+			assert.EqualError(t, err, "SyntaxError: /file.js: Unexpected character '\x00' (1:0)\n> 1 | \x00\n    | ^ at <eval>:2:28542(114)")
```
Comment (na--) [inline js/initcontext_test.go]: Same here, no need to match JS line numbers

File: js/lib/lib.go [hunk 3]
```diff
@@ -18,15 +18,19 @@
...
 )
 
-var CoreJS *goja.Program = goja.MustCompile(
-	"core-js/shim.js",
-	rice.MustFindBox("core-js").MustString("client/shim.js"),
-	true,
-)
+func GetCoreJS() *goja.Program {
+	return goja.MustCompile(
+		"core-js/shim.js",
```
Comment (na--) [inline js/lib/lib.go]: Not sure why, but `js/lib/rice-box.go` contains what seems like all of the files in the `core-js` folder when it should be (only?) the minified shim code. Running `grep -E "^\s+Filename:" js/lib/rice-box.go | wc -l` gives me `2036` files and the `rice-box.go` file itself is over 5 MB...

**Comment by na--** [comment]
It seems like `go.rice` doesn't make it easy to check if all generated files are up to date. My idea was to run `go generate ./...` in the CI process and fail if that caused changes in the checked out git repo. But since the `FileModTime` is tracked and updated, it won't work...

Since this PR removes the post-build embedding of static resources in the binary, we can use another embedding tool besides `go.rice` for the job. There are plenty of othhers ([[1]](https://github.com/avelino/awesome-go#resource-embedding), [[2]](https://tech.townsourced.com/post/embedding-static-files-in-go/)), but I'm not sure it's worth it at this point to evaluate a bunch of other tools just for that check, so maybe an issue for the future? @robingustafsson?

**Comment by luizbafilho** [comment]
I don't think that it is worth move from `go.rice` just for a simple check like this, that will probably be done just once in a year. 

It's very unlikely that one forgets to run `go generate ./...` after updating the libraries, and if that happens, since we have code reviews being done, someone else will catch it.

Moving away it is not a trivial task and that time will be better spent in other places focusing on more important things.

File: js/lib/lib.go (hunk 3 above)
Comment (luizbafilho) [reply js/lib/lib.go]: looking at the `shim.js` it is just a bunch of `require` I guess `go.rice` understand it and import every file that is required.

Comment (na--) [reply js/lib/lib.go]: Hmm I'm not sure if go.rice works like that. In any case, embedding only the minified version would get us a smaller git repo and binary and probably faster parsing by goja.

**Comment by luizbafilho** [comment]
@na-- changes made, now importing just the minified shim.

**Comment by na--** [comment]
Just a reminder, squash this PR before merging, since we don't want to pollute the git repo with all of the huge files in the intermediary commits

**Pull request merged**